docs/source/user_guide/installation/index.rst

```diff
@@ -50,8 +50,165 @@ Following GNU/Linux and macOS solvers are optionally available:
+`black <https://black.readthedocs.io/en/stable/>`__                              -                  examples           For code formatting.
+`jupyter <https://docs.jupyter.org/en/latest/>`__                                -                  examples           For example notebooks rendering.
+================================================================================ ================== ================== ===================================================================================================================================================================================
+
+.. _install.dev_dependencies:
+
+Dev dependencies
+^^^^^^^^^^^^^^^^^
+
+Installable with ``pip install "pybamm[dev]"``
+
+================================================================================ ================== ================== ===================================================================================================================================================================================
+Dependency                                                                       Minimum Version    pip extra          Notes
+================================================================================ ================== ================== ===================================================================================================================================================================================
+`numpy <https://numpy.org/>`__                                                   1.16.0             dev                For complex computation handling.
+`scipy <https://docs.scipy.org/doc/scipy>`__                                     2.8.2              dev                Provides algorithms for optimization, integration, interpolation etc.
+`pandas <https://pandas.pydata.org/docs/>`__                                     0.24.0             dev                For data manipulation & analysis.
+`anytree <https://anytree.readthedocs.io/en/latest/intro.html>`__                2.4.3              dev                For Tree Data Structure
+`autograd <https://autograd.readthedocs.io/en/latest/>`__                        1.2.0              dev                To differentiate python & numpy code.
+`scikit-fem <https://scikit-fem.readthedocs.io/en/latest/>`__                    0.2.0              dev                For performing finite element assembly.
+`imageio <https://imageio.readthedocs.io/en/stable/>`__                          2.9.0              dev                Handles diverse image data formats, including animated, volumetric, and scientific formats.
+`matplotlib <https://matplotlib.org/stable/>`__                                  2.0.0              dev                To plot various battery models, and analyzing battery performance.
+`casadi <https://web.casadi.org/docs/>`__                                        3.6.0              dev                For nonlinear optimization and algorithmic differentiation
```

Comment:
not dev

Reply:
`scikit-fem` has to be in `[dev]` or any other recommended section as we are not installing it elsewhere.

Reply:
Should not be in `[dev]`. It should have its own section I think.

Reply:
@tinosulzer should I make a separate section for `autograd`, `anytree` & `scikits-fem`, if so what should be the name I may give it?